physlr/physlr.py

```diff
@@ -1021,107 +1020,118 @@ def assign_read_molecule(mxs, moltomxs, mol_counts, bx):
         return "_" + str(random.choice(max_hits)), 0, 1
 
     @staticmethod
-    def determine_molecules_biconnected_components(g, u):
-        "Separate bi-connected components."
-        cut_vertices = set(nx.articulation_points(g.subgraph(g.neighbors(u))))
-        components = list(nx.connected_components(g.subgraph(set(g.neighbors(u)) - cut_vertices)))
+    def detect_communities_biconnected_components(g, node_set):
+        """Separate bi-connected components. Return components."""
+        cut_vertices = set(nx.articulation_points(g.subgraph(node_set)))
+        components = list(nx.connected_components(g.subgraph(node_set - cut_vertices)))
         components.sort(key=len, reverse=True)
-        return u, {v: i for i, vs in enumerate(components) if len(vs) > 1 for v in vs}
-
-    @staticmethod
-    def determine_molecules_k_clique_communities(g, u):
-        "Apply k-clique community detection algorithm after extracting bi-connected components."
-        cut_vertices = set(nx.articulation_points(g.subgraph(g.neighbors(u))))
-        components = list(nx.connected_components(g.subgraph(set(g.neighbors(u)) - cut_vertices)))
-        components.sort(key=len, reverse=True)
-        communities = []
-        for comp in components:
-            if len(comp) > 1:
-                communities += list(nxcommunity.k_clique_communities(g.subgraph(comp), 3))
-        return u, {v: i for i, vs in enumerate(communities) if len(vs) > 1 for v in vs}
+        return components
 
     @staticmethod
-    def determine_molecules_louvain(g, u):
-        "Apply louvain community detection algorithm after extracting bi-connected components."
-        import community as louvain
+    def detect_communities_k_clique(g, node_set, k=3):
+        """Apply k-clique community detection. Return communities."""
+        from networkx.algorithms import community as nxcommunity
 
-        cut_vertices = set(nx.articulation_points(g.subgraph(g.neighbors(u))))
-        components = list(nx.connected_components(g.subgraph(set(g.neighbors(u)) - cut_vertices)))
-        components.sort(key=len, reverse=True)
-        communities = []
-        for comp in components:
-            if len(comp) > 1:
-                partition = louvain.best_partition(g.subgraph(comp))
-                for com in set(partition.values()):
-                    list_nodes = [nodes for nodes in partition.keys() if partition[nodes] == com]
-                    if len(list_nodes) > 1:
-                        communities.append(list_nodes)
-        return u, {v: i for i, vs in enumerate(communities) if len(vs) > 1 for v in vs}
+        if len(node_set) > 1:
+            return [set(i) for i in nxcommunity.k_clique_communities(g.subgraph(node_set), k)]
```

Comment:
```suggestion
          return list(nx.community.k_clique_communities(g.subgraph(node_set), k))
```

Reply:
I wanted to share this issue with you @sjackman .
If I use my own code, the result will be in this format:
`[{8, 9, 10, 11, 13}, {4, 5, 6, 7, 13}, {0, 1, 2, 3}]`
and if I use yours, and I used to do the thing the way you do, the result will be in this format:
```[frozenset({8, 9, 10, 11, 13}),
 frozenset({4, 5, 6, 7, 13}),
 frozenset({0, 1, 2, 3})]```
and we will have some issue calling some other functions. Not sure where, but I faced this issue and I decided to change it this way. Any other ideas on how to solve the issue?

Reply:
So should I let it remains as it is now @sjackman?

Reply:
I'd prefer to use `frozenset` wherever possible. Let's use the code I suggested unless you can point me to the particular error. When I know what the error is, we can figure out the best way to address it.

Reply:
sure

Reply:
It's working properly with your change :D